### PR TITLE
(#336) Ensure tasks from non-production environments actually work

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

**Comment by treydock** [comment]
In draft while I validate localy.

**Comment by treydock** [comment]
So far local tests are looking good just want to run full playbook test which will be a bit while I fix up Choria certs.

```
{"@version":"1","@timestamp":"2023-12-27T17:54:54.767Z","clientip":"10.184.32.35","auth":"-","verb":"GET","requestprotocol":"HTTP/1.1","rawrequest":"GET /puppet/v3/tasks/std_tools/package_update?environment=dev_orchestration_caas_7757 HTTP/1.1","response":200,"bytes":1200,"total_service_time":223,"request":"http://OMIT:8140/puppet/v3/tasks/std_tools/package_update","referrer":"-","agent":"Choria version 0.19.0 http://choria.io","request.host":"OMIT:8140","request.accept":"application/json","request.accept-encoding":"gzip;q=1.0,deflate;q=0.6,identity;q=0.3","request.connection":"close","puppet.client-verify":"-","puppet.client-dn":"-","puppet.client-cert":"-","response.content-type":"application/json;charset=utf-8","response.content-length":"null","response.server":"null","response.connection":"close"}
```

The branch `dev_orchestration_caas_7757` was correctly used.

**Comment by treydock** [comment]
This is now working:

```
$ cat ../input.json
{
  "filter": ["networking.hostname=OMIT"],
  "tasks_environment": "dev_orchestration_caas_7757",
  "packages": "openssl",
  "check": true,
  "results_path": "/home/tdockendorf/results.json"
}

$ mco playbook run std_tools::package_update --modulepath site-modules:modules --input @../input.json
Notice: Scope(<module>/std_tools/plans/package_update.pp, 29): Discovered 1 node(s) in node set task_nodes
Info: Scope(<module>/choria/plans/tasks/download_files.pp, 14): Downloading files for task 'std_tools::package_update' onto 1 nodes
Notice: Scope(<module>/choria/plans/tasks/download_files.pp, 16): About to run task: mcollective task
Notice: Scope(<module>/choria/plans/tasks/download_files.pp, 16): Starting request for bolt_tasks#download against 1 nodes
Notice: Scope(<module>/choria/plans/tasks/download_files.pp, 16): Successful request c56b948a7c755163979e1a530f6d92c2 for bolt_tasks#download in 1.28s against 1 node(s)
Info: Scope(<module>/choria/plans/tasks/run.pp, 47): Running task 'std_tools::package_update' on 1 nodes
Notice: Scope(<module>/choria/plans/tasks/run.pp, 55): About to run task: mcollective task
Notice: Scope(<module>/choria/plans/tasks/run.pp, 55): Starting request for bolt_tasks#run_and_wait against 1 nodes
Notice: Scope(<module>/choria/plans/tasks/run.pp, 55): Successful request 980a638c054f5bd0a5c1694718501355 for bolt_tasks#run_and_wait in 45.99s against 1 node(s)
Notice: Scope(<module>/std_tools/plans/package_update.pp, 70): Writing results to: results.json

Playbook std_tools::package_update ran in 48.58 seconds: OK

Result:

     {
       "OMIT": {
         "updates": [
           {
             "package": "openssl.x86_64",
             "available_version": "1:1.1.1k-12.el8_9",
             "repository": "rhel-8-baseos"
           }
         ]
       }
     }
```

**Comment by treydock** [comment]
@ripienaar This is now ready for review.  Be great if this made it into next set of releases.

**Comment by ripienaar** [comment]
Unfortunately I think the tests are just broken now, no doubt a big push to fix it, not related to this PR.

File: lib/puppet/functions/choria/tasks/metadata.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@
 Puppet::Functions.create_function(:"choria::tasks::metadata") do
   dispatch :fetch do
     param "String", :task
+    optional_param "String", :environment
```
[the remark this answers is not in the document]
Comment (treydock) [reply lib/puppet/functions/choria/tasks/metadata.rb]: Yep, that seems reasonable.  I updated this to be `tasks_environment.

**Pull request merged**